src/anim/controller/anim-blend-tree-2d-cartesian.js

```diff
@@ -0,0 +1,51 @@
+import { Vec2 } from '../../math/vec2.js';
+import { math } from '../../math/math.js';
+
+import { AnimBlendTree } from './anim-blend-tree.js';
+
+/**
+ * @private
+ * @class
+ * @name AnimBlendTreeCartesian2D
+ * @classdesc An AnimBlendTree that calculates its weights using a 2D Cartesian algorithm
+ * @description Create a new BlendTree1D.
+ */
+class AnimBlendTreeCartesian2D extends AnimBlendTree {
+
+    static _p = new Vec2();
+
+    static _pip = new Vec2();
+
+    pointDistanceCache(i, j) {
+        var pointKey = `${i}${j}`;
+        if (!this._pointCache[pointKey]) {
+            this._pointCache[pointKey] = this._children[j].point.clone().sub(this._children[i].point);
+        }
+        return this._pointCache[pointKey];
+    }
+
+    calculateWeights() {
+        if (this.updateParameterValues()) return;
+        var i, j, pi, pipj, minj, result, weightSum;
+        AnimBlendTreeCartesian2D._p.set(...this._parameterValues);
```

Comment:
please double check what this looks like in ES5, we have not tested this yet.

Reply:
This is converted to a call to the apply method in es5.